### PR TITLE
Unify LinkageInfo word initialization

### PR DESCRIPTION
**Pull request by dsouzai**
The Linkage Info word is a 32-bit piece of memory right before the start PC of a compiled body that holds flags describing the body (sampling body, etc) as well as the offset describing the distance between the the Interpreter Entry Point and the JIT Entry Point. The initialization of this word has been done in a very raw manner, dealing directly with the memory rather than using the `J9::PrivateLinkage::LinkageInfo` abstraction. This PR cleans up this code.

Depends on https://github.com/eclipse/omr/pull/5583

### CODE REVIEW

**Comment by dsouzai** [comment]
@fjeremic could you please review?

**Comment by fjeremic** [comment]
> @fjeremic could you please review?

I see this is a breaking change. I'm going to be away for about 10 days starting Thursday so I may not be the best person to shepherd this one, unless you are willing to wait. Calling on @0xdaryl to see if he is up for it.

**Comment by 0xdaryl** [comment]
Breaking change?  Sign me up!

File: runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp [hunk 1]
```diff
@@ -1540,27 +1540,8 @@ void J9::ARM64::PrivateLinkage::performPostBinaryEncoding()
                    "Expecting the return type info instruction to be created");
 
    TR::ARM64ImmInstruction *linkageInfoWordInstruction = cg()->getReturnTypeInfoInstruction();
-   uint32_t linkageInfoWord = linkageInfoWordInstruction->getSourceImmediate();
-
-   intptr_t jittedMethodEntryAddress = reinterpret_cast<intptr_t>(getJittedMethodEntryPoint()->getBinaryEncoding());
-   intptr_t interpretedMethodEntryAddress = reinterpret_cast<intptr_t>(getInterpretedMethodEntryPoint()->getBinaryEncoding());
-
-   linkageInfoWord = (static_cast<uint32_t>(jittedMethodEntryAddress - interpretedMethodEntryAddress) << 16) | linkageInfoWord;
+   uint32_t linkageInfoWord = self()->initializeLinkageInfo(linkageInfoWordInstruction->getBinaryEncoding());
```
Comment (0xdaryl) [inline runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp:1543]: I don't think this is correct.  `initializeLinkageInfo` is in the `CodeGenerator` class, not `Linkage`.

Comment (dsouzai) [reply runtime/compiler/aarch64/codegen/ARM64PrivateLinkage.cpp:1543]: Ah you're right; I didn't realize this was the private linkage class. Good catch.

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#5583

**Comment by dsouzai** [comment]
The failed tests are due to Jenkins issues; might have to re-run them when appropriate.

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#5583

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#5583

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#5583

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8 win32

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#5583

**Comment by 0xdaryl** [comment]
I'm waiting on the results of some broader Java testing that I initiated within IBM before merging this change.

The Windows failures in the CI testing are infrastructural.  Previous CI testing in this PR passed the Windows tests.

**Comment by 0xdaryl** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#5583

**Review by 0xdaryl — APPROVED** [review]

**Comment by 0xdaryl** [comment]
This PR and its dependent OMR PR have had further tests run against it within IBM that all were successful.

**Pull request merged**